decode.go

```diff
@@ -29,6 +29,22 @@ func NewDecoder(r io.Reader) *Decoder {
 	return dec
 }
 
+// NewDecoderSize returns a new decoder that reads from r.
+//
+// The decoder introduces its own buffering and may read data from r beyond
+// the logfmt records requested.
+// The size argument specifies the size of the initial buffer that the
+// Decoder will use to read records from r.
+func NewDecoderSize(r io.Reader, size int) *Decoder {
+	buf := make([]byte, 0, size)
+	scanner := bufio.NewScanner(r)
+	scanner.Buffer(buf, size)
```

Comment:
`buf` is only used here, so to simplify let's inline the call to `make` and eliminate the `buf` variable.